### ISSUE TITLE
Links on the page open in the iframe

### ISSUE THREAD

**Issue by MethodGrab** [MethodGrab/firefox-custom-new-tab-page#1]
Clicking links on the chosen new tab page will open them inside the iframe not the top frame unless the links have the `target="_top"` attribute.

AFAIK there's no work around for this unless you control the site that is being used. If you do control it, you can add `target="_top"` to each link or include `<base target="_top">` in the `<head>` to apply it to all links.

**Comment by RandomEngy** [comment]
Can't extensions mess around with the DOM on the page? Otherwise how would GreaseMonkey work? Or is that not possible in this circumstance?

**Comment by MethodGrab** [comment]
You're right, it seems this might be possible with the [webRequest API](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/webRequest). I'll play around with the API a bit and see if it can be done.

**Comment by RandomEngy** [comment]
I was hoping it would be as simple as mucking with the already loaded elements. Loading from the WebRequest API sounds a lot harder because you have all the JS/CSS/XHR to deal with. Anyhow thanks for looking into it!

**Comment by Park0** [comment]
Would it be possible to do this with javascript? https://stackoverflow.com/questions/6158903/need-to-break-out-of-iframe-after-content-in-iframe-is-submitted by monitoring the location of the iframe and redirect the main page when it is linked to another domain.
I have not yet tried it myself.

**Comment by MethodGrab** [comment]
I've made a proof of concept that seems to work!
It's a bit hacky so I'll tidy it up and create a PR for people to test it out when I can.

**Comment by MethodGrab** [comment]
The experimental option to fix this (#14) is published on AMO as v1.0.0.

**Comment by Park0** [comment]
Does the page need a special target or something to get it to work? I updated the plugin. Enabled the option in the settings. Accepted the extra permission. But it still does not open in the same tab for me.

Thanks for working on this your add-on is great!

**Comment by mrx23dot** [comment]
This addon correctly opens the links without iframes. Could we copy that part?

![image](https://github.com/MethodGrab/firefox-custom-new-tab-page/assets/4163396/7c2a3a7a-1611-4d3e-a635-5210416a7ae9)
